src/IssueViz/TableControls/VSTableEventProcessorProvider.cs

```diff
@@ -0,0 +1,142 @@
+    [Name(nameof(VSTableEventProcessorProvider))]
+    [Order(Before = Priority.Default)]
+    // We can specify multiple data source types and data source i.e. the same provider can be used for
+    // multiple table controls/sources
+    [DataSourceType(StandardTableDataSources.ErrorTableDataSource)]
+    [DataSource(SonarLintTableControlConstants.ErrorListDataSourceIdentifier)]
+    internal class VSTableEventProcessorProvider : ITableControlEventProcessorProvider
+    {
+        private readonly IIssueTablesSelectionMonitor issueTableSelectionMonitor;
+
+        [ImportingConstructor]
+        internal VSTableEventProcessorProvider(IIssueTablesSelectionMonitor issueTableSelectionMonitor)
+        {
+            this.issueTableSelectionMonitor = issueTableSelectionMonitor;
+        }
+
+        ITableControlEventProcessor ITableControlEventProcessorProvider.GetAssociatedEventProcessor(IWpfTableControl tableControl)
+        {
+            if (tableControl == null)
+            {
+                return null;
+            }
+
+            return new EventProcessor(tableControl, issueTableSelectionMonitor);
+        }
+
+        /// <summary>
+        /// Filters and processes events from a specific WPF table, raising the higher-level "selected issue changed"
+        /// changed when appropriate.
+        /// </summary>
+        internal /* for testing */ class EventProcessor : ITableControlEventProcessor
+        {
+            private readonly IWpfTableControl wpfTableControl;
+            private readonly IIssueTablesSelectionMonitor selectionMonitor;
+
+            public EventProcessor(IWpfTableControl wpfTableControl, IIssueTablesSelectionMonitor errorListSelectionMonitor)
+            {
+                this.wpfTableControl = wpfTableControl;
+                this.selectionMonitor = errorListSelectionMonitor;
+            }
+
+            void ITableControlEventProcessor.PostprocessSelectionChanged(TableSelectionChangedEventArgs e)
+            {
+                try
+                {
+                    IAnalysisIssue selectedIssue = null;
+
+                    if (wpfTableControl.SelectedEntries.Count() == 1 &&
+                        wpfTableControl.SelectedEntry.TryGetSnapshot(out var snapshot, out var index) &&
+                        snapshot.TryGetValue(index, SonarLintTableControlConstants.IssueColumnName, out var issueObject) &&
+                        issueObject is IAnalysisIssue issueFromTable)
+                    {
+                        selectedIssue = issueFromTable;
+                    }
+
+                    selectionMonitor.SelectionChanged(selectedIssue);
+                }
+                catch(Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+                {
+                    // No-op
+                }
+            }
+
+            #region Event callbacks we are not interested in
+            void ITableControlEventProcessor.KeyDown(System.Windows.Input.KeyEventArgs args) { /* no-op */ }
```

Comment:
The code coverage is low because none of these unused methods are tested. I don't think it's worth adding tests for them, though.